pom.xml

```diff
@@ -263,6 +263,22 @@
             <compilerArg>-Xlint:all</compilerArg>
           </compilerArgs>
         </configuration>
+        <executions>
+          <execution>
+            <id>compile-java21</id>
+            <phase>compile</phase>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+            <configuration>
+              <compileSourceRoots>
+                <compileSourceroot>${project.basedir}/src/main/java21</compileSourceroot>
+              </compileSourceRoots>
+              <!-- cannot use multireleaseOutput, because we are creating a Java 8 class file -->
+              <outputDirectory>${project.build.outputDirectory}/META-INF/versions/21</outputDirectory>
```

Comment:
I get a warning about this
```
Parameter 'outputDirectory' is read-only, must not be used in configuration
```

Reply:
Ah, this is the one you mentioned (https://github.com/groovy/GMavenPlus/pull/287#issuecomment-1773766199).